File: Daily Challanges/2026-04-April/2026-04-07.py
def palindrome_locator(s):
    n = len(s)
    for i in range(n//2):
        if s[i] != s[n-i-1]:
            return "none"
    if n%2==0:
        return s[n//2-1] + s[n//2]
    else:
        return s[n//2]

print(palindrome_locator("racecar"))
print(palindrome_locator("level"))
print(palindrome_locator("freecodecamp"))

File: Daily Challanges/2026-04-April/2026-04-07.js
function palindromeLocator(str) {
  const n = str.length;
  const mid = Math.floor(n / 2);
  const reverse = str.split("").reverse().join("");

  if (str !== reverse) return "none";

  if (n % 2 === 0) return str[mid - 1] + str[mid];
  return str[mid];
}

console.log(palindromeLocator("racecar"));
console.log(palindromeLocator("level"));
console.log(palindromeLocator("freecodecamp"));
console.log(palindromeLocator("noon"));
